components/sidebar/sidebar.jsx

```diff
@@ -0,0 +1,692 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import $ from 'jquery';
+
+import React from 'react';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap';
+import ReactDOM from 'react-dom';
+import {FormattedMessage} from 'react-intl';
+import {browserHistory} from 'react-router';
+import {PropTypes} from 'prop-types';
+
+import {trackEvent} from 'actions/diagnostics_actions.jsx';
+import AppDispatcher from 'dispatcher/app_dispatcher.jsx';
+
+import * as ChannelUtils from 'utils/channel_utils.jsx';
+import {ActionTypes, Constants} from 'utils/constants.jsx';
+import * as Utils from 'utils/utils.jsx';
+
+import favicon from 'images/favicon/favicon-16x16.png';
+import redFavicon from 'images/favicon/redfavicon-16x16.png';
+
+import MoreChannels from 'components/more_channels';
+import MoreDirectChannels from 'components/more_direct_channels';
+
+import NewChannelFlow from '../new_channel_flow.jsx';
+import SidebarHeader from '../sidebar_header.jsx';
+import UnreadChannelIndicator from '../unread_channel_indicator.jsx';
+
+import SidebarChannel from './sidebar_channel';
+
+export default class Sidebar extends React.PureComponent {
+    static propTypes = {
+        config: PropTypes.object.isRequired,
+        publicChannelIds: PropTypes.array.isRequired,
+        privateChannelIds: PropTypes.array.isRequired,
+        favoriteChannelIds: PropTypes.array.isRequired,
+        directAndGroupChannelIds: PropTypes.array.isRequired,
+        unreadChannelIds: PropTypes.array.isRequired,
+        allChannels: PropTypes.object.isRequired,
+        currentChannel: PropTypes.object,
+        currentTeammate: PropTypes.object,
+        currentTeam: PropTypes.object.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        memberships: PropTypes.object.isRequired,
+        unreads: PropTypes.object.isRequired,
+        isSystemAdmin: PropTypes.bool.isRequired,
+        isTeamAdmin: PropTypes.bool.isRequired
```

Comment:
Can we add some comments to the props? It's a big component and having some comments on these will make it easier for someone to parse :)

Reply:
Done